QModManager/API/ModLoading/QModPatchMethod.cs

```diff
@@ -0,0 +1,37 @@
+﻿namespace QModManager.API.ModLoading
+{
+    using System;
+    using Advanced;
+
+    /// <summary>
+    /// Identifies a patch method for a QMod.<para/>
+    /// ALERT: The class that defines this method must have a <seealso cref="QModCoreInfo"/> attribute.
+    /// </summary>
+    /// <seealso cref="System.Attribute" />
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false, Inherited = false)]
+    public class QModPatchMethod : Attribute
```

Comment:
Maybe it's time for that `PreInit`/`Init`/`PostInit` type of initialization

Reply:
Already working on it.  
See `QModManager/API/ModLoading/Advanced/MetaPatchOrder.cs`

Reply:
That is a replacement for `Priority`, and not what I mean

Reply:
PreInit = Patch Early  
Init = Patch Normal  
PostInit = Patch Late

Reply:
Resolving for now as this is still WiP until the `Patcher` class is redone.  
There are 3 separate attributes: 
- `[QModPatchMethod]`, for 99.9% of mods
- `[QModPostPatchMethod]`, for mods that need something done _after_ all normal patching has completed
- `[QModPrePatchMethod]`, for mods that need something done _before_ all normal patching starts

Reply:
But mods can have both 3 at the same time i assume

Reply:
Yes. A mod could have one of each. And I will be adding some additional error checking for it.

For example:  
- Any mod with optional debug logging could use `[QModPrePatchMethod]` to set up logging preferences, then normal patching happens on `[QModPatchMethod]`.
- CustomCraft2 could set up logging preferences in either `[QModPatchMethod]` or `[QModPrePatchMethod]`, because it would do all its real patching in `[QModPostPatchMethod]` to avoid having to keep adding a `[LoadBefore]` attribute for every other mod.

Note that anything executing in `[QModPostPatchMethod]` would still run _before_ the SMLHelper patching.